hnn_core/opt_toy_example/generalized_optimization.py

```diff
@@ -0,0 +1,134 @@
+error_values_cobyla = list()
+
+def reduced_network():
+    # load default params
+    hnn_core_root = op.dirname(hnn_core.__file__)
+    params_fname = op.join(hnn_core_root, 'param', 'default.json')
+    params = read_params(params_fname) 
+    
+    # update params
+    params.update({'N_pyr_x': 3, 'N_pyr_y': 3})
+    net_obj = jones_2009_model(params, add_drives_from_params=True)
+    net_obj.clear_drives()
+    return net_obj
+
+def sim_dipole(net_obj, mu):
+    tstop = 10.
+    n_trials = 1
+    synaptic_delays = {'L2_basket': 0.1, 
+                       'L2_pyramidal': 0.1,
+                       'L5_basket': 1., 
+                       'L5_pyramidal': 1.}
+    weights_ampa = {'L2_basket': 0.5,
+                    'L2_pyramidal': 0.5,
+                    'L5_basket': 0.5,
+                    'L5_pyramidal': 0.5}
+    net_obj.add_evoked_drive('evprox1', 
+                             mu=mu, 
+                             sigma=2, 
+                             numspikes=1, 
+                             location='proximal', 
+                             weights_ampa=weights_ampa,
+                             synaptic_delays=synaptic_delays)
+    dpl = simulate_dipole(net_obj, tstop=tstop, n_trials=n_trials)[0] 
+    return dpl
+    
+def calculate_rmse(mu_predicted):
+    dpl_estimate = sim_dipole(net.copy(), mu_predicted)
+    rmse = np.sqrt( ((dpl_estimate.data['agg'] - dpl_true.data['agg'])**2).sum() / len(dpl_estimate.times) )
+    error_values_cobyla.append(rmse)
+    print('New mu: ', mu_predicted, ' RMSE: ', rmse)
+    return rmse
+
+    
+net = reduced_network()
+mu_true = 2
+mu_offset = 4
+dpl_true = sim_dipole(net.copy(), mu_true)
+dpl_offset = sim_dipole(net.copy(), mu_offset)    
+max_iter = 15
+
+#----------------------------------------------Bayesian optimization------------------------------------     
+bayesian_optimization_results = gp_minimize(calculate_rmse,
+                                            [(0.01, 10.)],
+                                            acq_func='EI',
+                                            n_calls=max_iter,
+                                            x0=[mu_offset],
+                                            random_state=64)    
+
+mu_bayesian = bayesian_optimization_results.x[0]
+dpl_bayesian = sim_dipole(net.copy(), mu_bayesian)
+    
+print('Bayesian Optimization Results \n-----------------------------')
+print('True mu: ', mu_true, '\nOffset mu: ', mu_offset, '\nOptimized mu: ', mu_bayesian)
+# plot_convergence(bayesian_optimization_results)
+error_values_bayesian = [np.min(bayesian_optimization_results.func_vals[:i]) for i in range(1, max_iter + 1)]
+
+#--------------------------------------------COBYLA optimization--------------------------------------- 
+cons = list()
+x0 = list()
+cons.append(lambda x, idx=0: 10. - x[0])
+cons.append(lambda x, ind=0: x[0] - 0.01)
+x0.append(mu_offset)
+error_values_cobyla.clear()
+cobyla_optimization_results = fmin_cobyla(calculate_rmse,
```

Comment:
could we also add the current optimization function to the comparisons? It has an argument called `return_rmse` which you can plot ...